tools/cli/lib/parserepoqueryargs.c

```diff
@@ -0,0 +1,205 @@
+                                             &pRepoqueryArgs->pszFile);
+                BAIL_ON_CLI_ERROR(dwError);
+            }
+            else if (strcasecmp(pSetOpt->pszOptName, "whatdepends") == 0)
+            {
+                dwError = TDNFSplitStringToArray(pSetOpt->pszOptValue,
+                    ",",
+                    &pRepoqueryArgs->ppszWhatDepends);
+                BAIL_ON_CLI_ERROR(dwError);
+            }
+            else if (strcasecmp(pSetOpt->pszOptName, "changelogs") == 0)
+            {
+                pRepoqueryArgs->nChangeLogs = 1;
+            }
+            else if (strcasecmp(pSetOpt->pszOptName, "depends") == 0)
+            {
+                pRepoqueryArgs->nDepends = 1;
+            }
+            else if (strcasecmp(pSetOpt->pszOptName, "requires-pre") == 0)
+            {
+                pRepoqueryArgs->nRequiresPre = 1;
+            }
+            else if (strcasecmp(pSetOpt->pszOptName, "available") == 0)
+            {
+                pRepoqueryArgs->nAvailable = 1;
+            }
+            else if (strcasecmp(pSetOpt->pszOptName, "installed") == 0)
+            {
+                pRepoqueryArgs->nInstalled = 1;
+            }
+            else if (strcasecmp(pSetOpt->pszOptName, "extras") == 0)
+            {
+                pRepoqueryArgs->nExtras = 1;
+            }
+            else if (strcasecmp(pSetOpt->pszOptName, "duplicates") == 0)
+            {
+                pRepoqueryArgs->nDuplicates = 1;
+            }
+            else if (strcasecmp(pSetOpt->pszOptName, "list") == 0)
+            {
+                pRepoqueryArgs->nList = 1;
+            }
+            else if (strcasecmp(pSetOpt->pszOptName, "source") == 0)
+            {
+                pRepoqueryArgs->nSource = 1;
+            }
+            else if (strcasecmp(pSetOpt->pszOptName, "upgrades") == 0)
+            {
+                pRepoqueryArgs->nUpgrades = 1;
+            }
+            else
+            {
+                int i;
+                int nFound = 0;
+
+                for (i = 0; i < REPOQUERY_KEY_COUNT; i++)
+                {
+                    if (strcasecmp(pSetOpt->pszOptName, depKeys[i]) == 0)
+                    {
+                        pRepoqueryArgs->anDeps[i] = 1;
+                        nFound = 1;
+                    }
+                }
+                if (!nFound)
+                {
+                    for (i = 0; i < REPOQUERY_KEY_COUNT; i++)
+                    {
+                        if (strcasecmp(pSetOpt->pszOptName, whatKeys[i]) == 0)
+                        {
+                            dwError = TDNFSplitStringToArray(pSetOpt->pszOptValue,
+                                ",",
+                                &pRepoqueryArgs->pppszWhatKeys[i]);
+                            BAIL_ON_CLI_ERROR(dwError);
+                            nFound = 1;
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
reduce the indentation levels here, ideally it should not exceed 3 levels

Reply:
I don't think putting the inner levels into a function makes it any more readable. I will add comments to the closing brackets to make it easier to understand.